tfs/src/main/java/hudson/plugins/tfs/model/TeamGitStatus.java

```diff
@@ -36,19 +38,32 @@ public static TeamGitStatus fromRun(@Nonnull final Run<?, ?> run) {
         if (result == null) {
             status.state = GitStatusState.Pending;
             status.description = status.state.toString();
-        }
-        else {
+        } else {
             status.state = RESULT_TO_STATE.get(result);
             status.description = result.toString();
         }
+        final Job<?, ?> job = run.getParent();
+        status.description = job.getDisplayName() + run.getDisplayName() + ": " + status.description;
         status.targetUrl = run.getAbsoluteUrl();
-        final Job<?, ?> project = run.getParent();
-        final String runDisplayName = run.getDisplayName();
-        final String projectDisplayName = project.getDisplayName();
-        status.context = new GitStatusContext(runDisplayName, projectDisplayName);
+        status.context = getStatusContext(job);
         return status;
     }
 
+    public static TeamGitStatus fromJob(@Nonnull final Job job) {
+        final TeamGitStatus status = new TeamGitStatus();
+        status.state = GitStatusState.Pending;
+        status.description = "Jenkins Job " + job.getDisplayName() + " queued";
+        status.targetUrl = job.getAbsoluteUrl();
+        status.context = getStatusContext(job);
+        return status;
+    }
+
+    private static GitStatusContext getStatusContext(@Nonnull final Job job) {
+        final String instanceUrl = StringUtils.stripEnd(Jenkins.getInstance().getRootUrl(), "/");
+        final String projectDisplayName = job.getDisplayName();

```

Comment:
This should be getFullDisplayName instead.

Reply:
or probably not, because display names might not be stable. What we need is something that takes into account the directory structure that is possible with jenkins plugins.